packages/arb-token-bridge-ui/tests/e2e/specs/depositCctp.cy.ts

```diff
@@ -3,236 +3,273 @@
-              cy.findByText('Ethereum Local gas fee')
-                .parent()
-                .siblings()
-                .last()
-                .contains(zeroToLessThanOneETH)
-                .should('be.visible')
-              cy.findByText('Arbitrum Local gas fee')
-                .parent()
-                .siblings()
-                .last()
-                .contains(zeroToLessThanOneETH)
-                .should('be.visible')
-            })
+        // By default, confirm button is disabled
+        cy.findByRole('button', {
+          name: /Continue/i
         })
+          .should('be.visible')
+          .should('be.disabled')
 
-        context('should show clickable deposit button', () => {
-          cy.findByRole('button', {
-            name: /Move funds to Arbitrum Sepolia/i
-          })
-            .should('be.visible')
-            .should('be.enabled')
-            .click()
+        // Checkbox
+        cy.findByRole('switch', {
+          name: /I understand that I'll have to send/i
         })
+          .should('be.visible')
+          .click()
+        cy.findByRole('switch', {
+          name: /I understand that it will take/i
+        })
+          .should('be.visible')
+          .click()
 
-        context('Should display CCTP modal', () => {
-          // Tabs
-          cy.findByRole('tab', {
-            name: "Arbitrum's bridge (USDC.e)",
-            selected: true
-          }).should('exist')
-          cy.findByRole('tab', {
-            name: 'Third party (USDC)',
-            selected: false
-          }).should('exist')
-          cy.findByRole('tab', {
-            name: 'Circle (USDC)',
-            selected: false
-          })
-            .should('exist')
-            .click()
+        cy.findByRole('switch', {
+          name: /I understand USDC.e/i
+        })
+          .should('be.visible')
+          .click()
 
-          // By default, confirm button is disabled
-          cy.findByRole('button', {
-            name: 'Confirm'
-          })
-            .should('be.visible')
-            .should('be.disabled')
+        cy.findByRole('button', {
+          name: /Continue/i
+        })
+          .should('be.visible')
+          .should('be.enabled')
+          .click()
 
-          // Checkbox
-          cy.findByRole('switch', {
-            name: /I understand that I'll have to send/i
-          })
-            .should('be.visible')
-            .click()
-          cy.findByRole('switch', {
-            name: /I understand that it will take/i
-          })
-            .should('be.visible')
-            .click()
+        cy.findByText(/I understand that I have to/).click()
+        cy.findByRole('button', {
+          name: /Pay approval fee of/
+        }).click()
+        cy.log('Approving USDC...')
+        cy.confirmMetamaskPermissionToSpend(USDCAmountToSend.toString()).then(
+          () => {
+            // eslint-disable-next-line
+            cy.wait(40_000)
```

Comment:
Do we really need to wait 40s for the metamask popup?

Reply:
Anyway, we should have a more dynamic text on the transfer button imo.

E.g. if transferring, it should say "Moving funds to {networkName}".

If approving, it should say "Approving {tokenSymbol}".

This way we could target these elements and use `waitUntil` instead of arbitrary `wait`.

If you agree could we create a ticket?

Reply:
Sepolia is slow, so we could have it lower, but there would be too many e2e failing sadly.

Something like this would be nice? Maybe we could even reuse the transaction detail?
![image](https://github.com/OffchainLabs/arbitrum-token-bridge/assets/5085333/e9b05c72-349e-4426-a55e-35e9661d90bd)

Reply:
Yes, this would be ideal. Would need to loop in design team for this one.